internal/pkg/agent/install/flavors.go

```diff
@@ -0,0 +1,259 @@
+)
+
+const (
+	FlavorBasic   = "basic"
+	FlavorServers = "servers"
+
+	DefaultFlavor  = FlavorBasic
+	flavorFileName = ".flavor"
+)
+
+type SkipFn func(relPath string) bool
+
+var ErrUnknownFlavor = fmt.Errorf("unknown flavor")
+
+type FlavorDefinition struct {
+	Name       string
+	Components []string
+}
+
+func UsedFlavor(topPath, defaultFlavor string) (string, error) {
+	filename := filepath.Join(topPath, flavorFileName)
+	content, err := os.ReadFile(filename)
+	if err != nil {
+		// file does not exist, flavor was not marked probably due to earlier version
+		// fallback to default if defined
+		if defaultFlavor != "" && os.IsNotExist(err) {
+			return defaultFlavor, nil
+		}
+
+		// failed reading flavor, do not break behavior and apply none as widest
+		return "", err
+	}
+
+	return string(content), nil
+}
+
+func Flavor(detectedFlavor string, registryPath string, flavorsRegistry map[string][]string) (FlavorDefinition, error) {
+	if flavorsRegistry == nil {
+		f, err := os.Open(registryPath)
+		if err != nil {
+			return FlavorDefinition{}, err
+		}
+		manifest, err := v1.ParseManifest(f)
+		if err != nil {
+			return FlavorDefinition{}, err
+		}
+		defer f.Close()
+		flavorsRegistry = manifest.Package.Flavors
+	}
+
+	components, found := flavorsRegistry[detectedFlavor]
+	if !found {
+		return FlavorDefinition{}, ErrUnknownFlavor
+	}
+
+	return FlavorDefinition{detectedFlavor, components}, nil
+}
+
+// SpecsForFlavor returns spec files associated with specific flavor
+func SpecsForFlavor(flavor FlavorDefinition) []string {
+	specs := []string{}
+	for _, component := range flavor.Components {
+		specs = append(specs, fmt.Sprintf("%s.spec.yml", component))
+	}
+
+	return specs
+}
+
+// ApplyFlavor scans agent comonents directory and removes anything
+// that is not mapped and needed for currently used flavor
+func ApplyFlavor(versionedHome string, flavor FlavorDefinition) error {
+	skipFn, err := SkipComponentsPathFn(versionedHome, flavor)
+	if err != nil {
+		return err
+	}
+
+	componentsDir := filepath.Join(versionedHome, "components")
+	filesToRemove := []string{}
+
+	err = filepath.Walk(componentsDir, func(path string, info fs.FileInfo, err error) error {
+		if errors.Is(err, fs.ErrNotExist) {
+			return nil
+		}
+		if err != nil {
+			return fmt.Errorf("walk on %q failed: %w", componentsDir, err)
+		}
+
+		if skipFn != nil && skipFn(path) {
+			// remove as file is not needed
+			filesToRemove = append(filesToRemove, path)
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, ftr := range filesToRemove {
+		if removeErr := os.RemoveAll(ftr); !os.IsNotExist(removeErr) {
+			err = removeErr
+		}
+	}
+
+	return err
+}
+
+// ParseComponentFiles parses spec files and returns list of associated paths with component.
+// Default set consisting of binary, spec file and default config file is always present
+func ParseComponentFiles(content []byte, filename string, includeDefaults bool) ([]string, error) {
```

Comment:
nit: we already have a file encapsulating spec files, this should probably live there and not with the install logic https://github.com/elastic/elastic-agent/blob/af2a1c2cf9ccfb06c95ca51ec91fb5d83413e1e8/pkg/component/load.go#L241